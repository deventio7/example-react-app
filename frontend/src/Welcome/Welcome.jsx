// @flow strict
import React from 'react';

import WelcomeCss from './Welcome.scss';

const Welcome = () => {
    return (
        <div>
            <h1 className={WelcomeCss.title}>
                Landing Page Content
            </h1>
        </div>);
};

export default Welcome;
