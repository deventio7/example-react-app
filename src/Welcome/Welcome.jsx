import React from 'react';
import PropTypes from 'prop-types';
import WelcomeCss from './Welcome.scss';


const Welcome = ({ username }) => {
    return (
        <div>
            <h1 className={WelcomeCss.test}>hello {username}</h1>
        </div>);
};

Welcome.propTypes = {
    username: PropTypes.string.isRequired
};

export default Welcome;
