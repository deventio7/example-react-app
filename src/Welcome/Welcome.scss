@import '../constant';

.test {
    color: $my-yellow;
}
