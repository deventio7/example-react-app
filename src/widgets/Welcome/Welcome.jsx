// @flow

import React from 'react';
import { Icon } from 'antd';
// $FlowFixMe
import WelcomeCss from './Welcome.scss';

type Props = {
    username: string
};

const Welcome = (props: Props) => {
    return (
        <div>
            <Icon type="smile-o" />
            <h1 className={WelcomeCss.test}>hello {props.username}</h1>
        </div>);
};


export default Welcome;
