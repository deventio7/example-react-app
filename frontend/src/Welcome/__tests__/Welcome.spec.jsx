import React from 'react';
import { shallow } from 'enzyme';
import Welcome from '../Welcome';

describe('Welcome', () => {
    it('should create a snapshot', () => {
        const wrapper = shallow(<Welcome />);
        expect(wrapper).toMatchSnapshot();
    });
});
