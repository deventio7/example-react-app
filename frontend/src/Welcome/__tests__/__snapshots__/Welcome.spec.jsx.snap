// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should create a snapshot 1`] = `
<div>
  <h1>
    Landing Page Content
  </h1>
</div>
`;
